TrackletAlgorithm/Constants.h

```diff
@@ -55,27 +64,54 @@ constexpr double kphi0 = 1.54259e-05;
 constexpr double kt = 0.00195312;
 constexpr double kz0 = 0.0585938;
 
-// mean positions of layers and disks (the 1.0e-1 is a fudge factor needed to
-// get the floating point truncation right)
-constexpr int rmean[6] = {
-  static_cast<int>(24.9316 / kr + 1.0e-1),
-  static_cast<int>(37.1777 / kr + 1.0e-1),
-  static_cast<int>(52.2656 / kr + 1.0e-1),
-  static_cast<int>(68.7598 / kr + 1.0e-1),
-  static_cast<int>(86.0156 / kr + 1.0e-1),
-  static_cast<int>(108.311 / kr + 1.0e-1)
-};
-constexpr int zmean[5] = {
-  static_cast<int>(131.191 / kz + 1.0e-1),
-  static_cast<int>(154.98 / kz + 1.0e-1),
-  static_cast<int>(185.332 / kz + 1.0e-1),
-  static_cast<int>(221.602 / kz + 1.0e-1),
-  static_cast<int>(265.02 / kz + 1.0e-1)
-};
+// layer projection digitization constants
+constexpr double kphiproj456 = 7.71867e-06;
+constexpr double kphider = 8.23325e-06;
+constexpr double kzproj = 0.0585938;
+constexpr double kzder = 0.015625;
+
+// disk projection digitization constants
+constexpr double kphiprojdisk = 6.17494e-05;
+constexpr double kphiprojderdisk = 1.64665e-05;
+constexpr double krprojdisk = 0.0585938;
+constexpr double krprojderdisk = 0.0078125;
+
+// detector constants
+constexpr int N_LAYER = 6; // # of barrel layers assumed
+constexpr int N_DISK = 5; // # of endcap disks assumed
+constexpr double bfield = 3.8; // T
+constexpr int rmean[N_LAYER] = { 851, 1269, 1784, 2347, 2936, 3697 };
+constexpr int zmean[N_DISK] = { 2239, 2645, 3163, 3782, 4523 };
+constexpr double zlength = 120.0; // cm
+constexpr double rmindiskvm = 22.5; // cm
+constexpr double rmindisk = 20.0; // cm
+constexpr double rmaxdisk = 120.0; // cm
+constexpr double rmindisktc = 12.0; // cm
+constexpr double rmaxdisktc = 112.0; // cm
+
+// cut constants
+constexpr double ptcut = 1.91; // GeV
+constexpr double rinvcut = 0.01 * c * bfield / ptcut; // 0.01 to convert to 1/cm
+constexpr double z0cut = 15.0; // cm
+
+// various bit widths
+constexpr unsigned nbitsallstubs[N_LAYER + N_DISK] = {3, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2};
+constexpr unsigned int nbits_maxvm = 5; // number of bits needed for max number of VMs per layer/disk (max number is 32)
 
 // List of regions for memory template parameters
 enum regionType {BARRELPS, BARREL2S, BARRELOL, BARREL, DISKPS, DISK2S, DISK};
 
+namespace TF {
```

Comment:
@aehart I left these enums in a TF namespace to avoid confusion with more generic uses of seed, layer, disk. An alternative might be to change these enum names to something less prone to be accidentally reused. Also, there are now a slew of trivial changes to TC which change many TC:: to TF::

Reply:
I think this is fine for now. Our use of namespaces right now is a little ad-hoc and inconsistent, so at some point, we'll need to overhaul it. But we'll address that in a future issue.